abi/codecForCompositeTypes_option.go

```diff
@@ -0,0 +1,65 @@
+package abi
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+)
+
+func (c *codec) encodeNestedOption(writer io.Writer, value OptionValue) error {
+	if value.Value == nil {
+		_, err := writer.Write([]byte{0})
+		return err
+	}
+
+	_, err := writer.Write([]byte{1})
+	if err != nil {
+		return err
+	}
+
+	return c.doEncodeNested(writer, value.Value)
+}
+
+func (c *codec) encodeTopLevelOption(writer io.Writer, value OptionValue) error {
+	if value.Value == nil {
+		return nil
```

Comment:
This can happen if you have only one Option arg for endpoint and you send it empty, right?
Can we double check with someone from wasm team that in this case empty "" is correct as input?

All things considered.. using `return c.encodeNestedOption(writer, value)` here should work no matter what

Reply:
"" is the way we encode Option::None, indeed

Reply:
Thanks for the question & resolution.

For reference: https://docs.multiversx.com/developers/data/composite-values#options.